### PR TITLE
feat: Why Koda artist landing V1

### LINKED ISSUE #9992: Why Koda artist landing V1
# Koda page 
**gm, for better presentation, here is the design for our new "Why Kodadot" page** 

its almost static page, one cta in the footer 

question is, whether we should add navbar there, i think we could actually

image

# Design 
- added to our new page inside handoff file: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=9437-38301&mode=design&t=81TmqAnkshCLTRdp-11

## Top part
- its v1, goal is probably to have it ready, nothing fancy for now
- top part consists of nfts from our whitelisted collections (collection of kodaminter on ahp) 
- i was thinking if the top part could move - one row of the imgs going to left, the other to right - slow animation 
	- that can be for v2 tho 
- next thing is the overview and content on the page, currently looking like this: 
image

v2 could be that it will be clickable and it will scroll down to the section, up to you if you want to implement it right away 

## Content 
- one style, number, heading, paragraph
- please align it as it is in the figma 
- between the sections, there is divider 

### offline exhibition 
- this one is slightly different
- it has a gallery 
image
- underneath benefits 

## Footer 
image

call to action - will open telegram link on @JustLuuuu - don't know how its done, could you please post the link here? 👀 

# Other
- mobile have slightly different layout, content is similar 
- dark mode is following our rules of color pairs 
- from 1440 it does does not increase its width, stays in center as for example our gallery item etc. 
- please see the spacing

### LINKED ISSUE #10017: Add why koda link to footer
once merged 
- #9992 

please add "Why Koda?" to footer 
image

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #9992
- [x] Closes #10017

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-04-07 at 15 13 59@2x

CleanShot 2024-04-08 at 09 27 34@2x

### CODE REVIEW

File: components/why-koda/HeroSection.vue [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    return data?.value.events || []
+  },
+  {
+    transform: (events) =>
+      uniqBy(
+        events.map((event) => event.nft),
+        'id',
+      ).slice(0, NFTS_PER_STRIP * AMOUNT_OF_STRIPS),
+  },
+)
```
Comment (vikiival) [inline components/why-koda/HeroSection.vue:117]: Isn't it better t write usable resolver on the squid side?

File: components/why-koda/CtaSection.vue [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        class="!px-8 !py-10 md:!px-14 max-w-[740px] bg-background-color border rounded-xl shadow-primary">
+        <p class="text-3xl md:text-5xl font-bold capitalize">
+          {{ $t('whyKoda.cta.heading') }}
+        </p>
+        <p class="!mt-5 max-w-[510px] text-xl">
+          {{ $t('whyKoda.cta.description') }}
+        </p>
+
+        <NeoButton
+          :tag="NuxtLink"
```
Comment (daiagi) [inline components/why-koda/CtaSection.vue:14]: i believe the correct tag here is `a` (external link)

File: components/why-koda/HeroSection.vue [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      class="md:absolute top-0 right-0 md:w-[55%] h-full overflow-hidden bg-neutral-3 dark:bg-neutral-11 max-md:border-b md:border-l">
+      <div
+        class="max-md:py-9 flex flex-col max-md:gap-4 md:gap-8 h-full justify-center">
+        <div
+          v-for="(nfts, index) in nftsByStrip"
+          :key="`strip-${index}`"
+          class="flex max-md:gap-4 md:gap-8"
+          :class="{
+            'transform -translate-x-60': index !== 1,
+            'transform translate-x-4 md:translate-x-8': index === 1,
```
Comment (daiagi) [inline components/why-koda/HeroSection.vue:48]: maybe `first` or `last` can be your friend here?
https://tailwindcss.com/docs/hover-focus-and-other-states#first-last-odd-and-even

File: components/why-koda/HeroSection.vue [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import uniqBy from 'lodash/uniqBy'
+
+defineProps<{
+  sections: { name: string; id: string }[]
+}>()
+
+const NFTS_PER_STRIP = 10
+const AMOUNT_OF_STRIPS = 2
+
+const { isMobileOrTablet: isTouch } = useDevice()
```
Comment (daiagi) [inline components/why-koda/HeroSection.vue:77]: can be done with css instead?

File: components/why-koda/HeroSection.vue [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      ).slice(0, NFTS_PER_STRIP * AMOUNT_OF_STRIPS),
+  },
+)
+
+const nftsByStrip = computed(() => chunk(nftss?.value ?? [], NFTS_PER_STRIP))
+
+const logoSrc = computed(() => {
+  const variant = isTouch ? 'Koda' : 'Koda_logo'
+  const color = isDarkMode.value ? '_dark' : ''
+  return `/${variant}${color}.svg`
```
Comment (daiagi) [inline components/why-koda/HeroSection.vue:124]: can be achieved using visibility modifiers and 
https://tailwindcss.com/docs/dark-mode#basic-usage

File: components/why-koda/Sections.vue [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          :key="img"
+          class="h-[216px] min-w-[291px] md:h-[358px] md:min-w-[480px] shadow-primary border rounded-xl overflow-hidden">
+          <BasicImage :src="img" alt="offline_exhibitions" />
+        </div>
+      </div>
+
+      <div :class="containerClasses">
+        <div class="pb-20">
+          <p class="text-xl md:text-2xl">{{ $t('benefits') }}:</p>
+          <div class="flex gap-4 flex-wrap !mt-4">
```
Comment (daiagi) [inline components/why-koda/Sections.vue:30]: `!mt-4` -> `mt-4` ?

File: components/why-koda/Sections.vue [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          :id="lastSection.id"
+          class="py-14 md:py-[5.5rem] border-k-shade2 border-t">
+          <WhyKodaSectionItem :section="lastSection" />
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+<script lang="ts" setup>
+const containerClasses = 'container mx-auto max-md:px-4'
```
Comment (daiagi) [inline components/why-koda/Sections.vue:50]: odd choice

why not put it in `style` block?

File: components/why-koda/Sections.vue [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+const props = defineProps<{
+  sections: { name: string; id: string; description: string; number: number }[]
```
Comment (daiagi) [inline components/why-koda/Sections.vue:53]: type /  interface plz

File: components/why-koda/Sections.vue [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+}>()
+
+const glob = import.meta.glob('~/public/why-koda-*.webp', { eager: true })
+
+const offlineExhibitionsImages = Object.entries(glob).map(
+  ([, value]) => value.default,
+)
+
+const mainSections = computed(() =>
```
Comment (daiagi) [inline components/why-koda/Sections.vue:62]: seems fragile, reliant on order of the `sections` props

File: locales/en.json [hunk 10]
```diff
@@ -5,10 +5,12 @@
   "scheduled": "Scheduled",
   "supply": "Supply",
   "artist": "Artist",
+  "why": "why",
```
Comment (daiagi) [inline locales/en.json:8]: capital "Why" i guess ?

File: components/why-koda/Sections.vue [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  props.sections.slice(0, props.sections.length - 1),
+)
+
+const lastSection = computed(() => props.sections[props.sections.length - 1])
```
Comment (daiagi) [inline components/why-koda/Sections.vue:66]: for me, this does not require a computed ref
but I'm nit picking

File: locales/en.json [hunk 12]
```diff
@@ -1189,6 +1191,56 @@
...
+      },
+      "generative_art": {
+        "name": "Generative Art",
+        "description": "Our heart beats for generative art, where code meets canvas, crafting the extraordinary out of the ordinary. It's not just art; it's an experience, a journey through creativity powered by the elegance of algorithms"
+      },
+      "koda": {
+        "name": "Koda",
+        "description": "Koda means code in the Slovenian language. Which is now our primary focus. To deliver not only high-quality generative art but also to grow our code base, add new features and expand our horizons not only in the Polkadot ecosystem but also outside of it."
+      },
+      "weekly_drops": {
```
Comment (daiagi) [inline locales/en.json:1214]: usually we do camelCase

**Review by daiagi — CHANGES_REQUESTED** [review]

File: components/why-koda/CtaSection.vue (hunk 2 above)
Comment (hassnian) [reply components/why-koda/CtaSection.vue:14]: yep, fixed

File: components/why-koda/HeroSection.vue [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+          <div class="h-[calc(100%-13px)] flex items-center">
+            <img :src="logoSrc" alt="koda-logo" class="h-full" />
+          </div>
+        </h1>
+
+        <p class="!mt-6 text-xl">
+          {{ $t('whyKoda.embraceTheFutureOfGenerativeArt') }}
+        </p>
+
+        <div class="!mt-10 md:mt-16">
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/why-koda/HeroSection.vue:18]: see https://github.com/kodadot/nft-gallery/issues/9973

File: components/why-koda/HeroSection.vue (hunk 3 above)
Comment (hassnian) [reply components/why-koda/HeroSection.vue:48]: thanks!, fixed

File: components/why-koda/HeroSection.vue (hunk 4 above)
Comment (hassnian) [reply components/why-koda/HeroSection.vue:77]: yes, fixed

File: components/why-koda/HeroSection.vue (hunk 5 above)
Comment (hassnian) [reply components/why-koda/HeroSection.vue:124]: fixed

File: components/why-koda/Sections.vue (hunk 6 above)
Comment (hassnian) [reply components/why-koda/Sections.vue:30]: again https://github.com/kodadot/nft-gallery/issues/9973

File: components/why-koda/Sections.vue (hunk 8 above)
Comment (hassnian) [reply components/why-koda/Sections.vue:53]: fixed

File: locales/en.json (hunk 12 above)
Comment (hassnian) [reply locales/en.json]: fixed

File: components/why-koda/Sections.vue (hunk 9 above)
Comment (hassnian) [reply components/why-koda/Sections.vue]: fixed

File: components/why-koda/HeroSection.vue (hunk 1 above)
Comment (daiagi) [reply components/why-koda/HeroSection.vue:117]: agree

File: components/why-koda/Sections.vue (hunk 7 above)
Comment (hassnian) [reply components/why-koda/Sections.vue]: fixed bb1eb2b, the `container`class was conflicting with tailwindcss

File: components/why-koda/HeroSection.vue (hunk 1 above)
Comment (hassnian) [reply components/why-koda/HeroSection.vue:117]: on it

**Review by daiagi — APPROVED** [review]
apart from @vikiival  review
lgtm

**Comment by exezbcz** [comment]
hello sir 

nice job! 

few things: 
## should have a black stroke 
image
- could you please also remove the hover? 


## Buttons
image

thought you will make it a button like this one but with different dimensions 
image

**Comment by hassnian** [comment]
> hello sir
> 
> nice job!
> 
> few things:
> 
> ## should have a black stroke
> image
> 
> * could you please also remove the hover?
> 
> ## Buttons
> image
> 
> thought you will make it a button like this one but with different dimensions image

@exezbcz fixed

**Comment by exezbcz** [comment]
yup, works 

one last detail, the row is incomplete 

one nft is missing on bottom right place
image

**Review by exezbcz — APPROVED** [review]

**Comment by exezbcz** [comment]
letsgo, tablet and other sizes can be followup

thank you

File: components/why-koda/HeroSection.vue (hunk 1 above)
Comment (vikiival) [reply components/why-koda/HeroSection.vue:115]: this?

**Comment by prury** [comment]
https://github.com/kodadot/nft-gallery/pull/10016#discussion_r1559411064

File: components/why-koda/HeroSection.vue (hunk 1 above)
Comment (hassnian) [reply components/why-koda/HeroSection.vue]: https://github.com/kodadot/stick/pull/272

File: components/why-koda/HeroSection.vue [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+      limit: 12,
+      active: [true],
+      chain: ['ahp'],
+    }),
+  {
+    transform: (data) => data.map((drop) => drop.collection),
+  },
+)
+
+const { data: nftss } = await useAsyncData(async () => {
```
Comment (vikiival) [inline components/why-koda/HeroSection.vue:92]:
```suggestion
const { data: items } = await useAsyncData(async () => {
```

Comment (hassnian) [reply components/why-koda/HeroSection.vue]: fixed

**Comment by prury** [comment]
nft background no longer there: 

image

**Comment by hassnian** [comment]
> nft background no longer there:
> 
> image

code got updated https://github.com/kodadot/nft-gallery/pull/10016#discussion_r1559411064 

now https://github.com/kodadot/stick/pull/272 is needed to fetch the recent nfts

**Comment by exezbcz** [comment]
is only the resolver needed for this to be ready? 

if yes, we can probably hardcode some nfts and go on?

**Comment by vikiival** [comment]
Imo lets go without resolver :). The query is not something I wanna merge to indexer

**Comment by hassnian** [comment]
> nft background no longer there:

reverted changes , it's working again 

CleanShot 2024-04-17 at 08 05 28@2x

**Pull request merged**